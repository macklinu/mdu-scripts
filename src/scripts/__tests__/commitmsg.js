const {
  unquoteSerializer,
  winPathSerializer,
  projectRootSerializer,
} = require('./helpers/serializers')

expect.addSnapshotSerializer(unquoteSerializer)
expect.addSnapshotSerializer(winPathSerializer)
expect.addSnapshotSerializer(projectRootSerializer)

test('precommit executes validate-commit-msg with no args', () => {
  const mockExeca = require('execa')
  const utils = require('../../utils')
  Object.assign(utils, {
    resolveBin: (modName, { executable = modName } = {}) => executable,
  })
  process.exit = jest.fn()

  process.argv = ['node', '../commitmsg']

  try {
    // tests
    require('../commitmsg')
    expect(mockExeca).toHaveBeenCalledTimes(1)
    const [firstCall] = mockExeca.mock.calls
    const [script, calledArgs] = firstCall
    expect([script, ...calledArgs].join(' ')).toMatchSnapshot()
  } catch (error) {
    throw error
  } finally {
    // afterEach
    jest.resetModules()
  }
})
