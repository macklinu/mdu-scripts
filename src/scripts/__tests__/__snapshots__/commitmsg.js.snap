// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`precommit executes validate-commit-msg with no args 1`] = `validate-commit-msg`;
